File: src/main/java/sh/miles/cosmictools/stage/RunStage.java
package sh.miles.cosmictools.stage;

import joptsimple.OptionSet;
import org.jetbrains.annotations.NotNull;

import java.util.Map;

/**
 * Represents a stage within the run cycle
 */
public interface RunStage {

    /**
     * Runs the current stage
     *
     * @param options   the options
     * @param propagate mutable map to propagate different datum to other stages
     * @return the exit code or -1 for none
     */
    int runStage(@NotNull final OptionSet options, @NotNull final Map<String, Object> propagate) throws Exception;

}

File: src/main/kotlin/sh/miles/cosmictools/runstage/RunStage.kt
package sh.miles.cosmictools.runstage

import joptsimple.OptionSet

/**
 * An Abstract RunStage which represents a stage in the running of CosmicTools
 */
interface RunStage {
    /**
     * Executes the stage
     *
     * @param propagate a map of data that can be passed between stages
     */
    fun runStage(options: OptionSet, propagate: MutableMap<String, Any>): Int
}
